torch/_inductor/codegen/cpp.py

```diff
@@ -2494,8 +2523,18 @@ def reduction(self, dtype, src_dtype, reduction_type, value):
             dtype,
             reduction_combine_fn=self.reduction_combine_vec,
             reduction_init_fn=self.reduction_init_vec,
-            welford_weight_reciprocal_vec_fn=self.welford_weight_reciprocal_vec,
         )
+        if reduction_type == "welford_reduce":
+            # use masked acc_vec for tail vec kernel
+            self._gen_parallel_reduction_buffers(
+                f"masked_{acc_vec}",
+                acc_type_vec,
+                reduction_type,
+                dtype,
+                reduction_combine_fn=self.reduction_combine_vec,
+                reduction_init_fn=self.reduction_init_vec,
```

Comment:
Can we move this logic into if scope of https://github.com/pytorch/pytorch/pull/126526/files#diff-5ab7b0235e2076a5fc6629ba0b109208940f5b94f5c13babc3e0f87cf4fcec82R2488 to save an extra flag `self.gen_weight_recps`? The point is that CSE should already help to avoid duplication that we want to avoid with `self.gen_weight_recps`, right?

Reply:
Yes. I will move this logic into the if scope.